batch/batch/worker/worker.py

```diff
@@ -2212,25 +2213,31 @@ async def cleanup(self):
         assert self.worker.fs
         assert self.jvm
 
+        with self.step('uploading_log'):
+            log_contents = await self.worker.fs.read(self.log_file)
+            await self.worker.file_store.write_log_file(
+                self.format_version, self.batch_id, self.job_id, self.attempt_id, 'main', log_contents
+            )
+
         if self.cloudfuse:
             for config in self.cloudfuse:
                 if config['mounted']:
                     bucket = config['bucket']
                     assert bucket
                     mount_path = self.cloudfuse_data_path(bucket)
-                    await self.jvm.cloudfuse_mount_manager.unmount(mount_path, user=self.user, bucket=bucket)
-                    config['mounted'] = False
+                    try:
+                        await self.jvm.cloudfuse_mount_manager.unmount(mount_path, user=self.user, bucket=bucket)
+                        config['mounted'] = False
+                    except asyncio.CancelledError:
+                        raise
+                    except Exception:
+                        log.exception(f'error while unmounting cloudfuse for {self.jvm_name} for job {self.id}')
```

Comment:
Let's include the same information we have above:
```suggestion
                        log.exception(
                            f'while unmounting fuse blob storage {bucket} from {mount_path} for {self.jvm_name} for job {self.id}'
                        )
```